code/modules/mob/dead/new_player/preferences_setup.dm

```diff
@@ -46,6 +46,51 @@
 	if(randomise[RANDOM_NAME])
 		real_name = pref_species.random_name(gender,1)
 
+///Setup a hardcore random character and calculate their hardcore random score
+/datum/preferences/proc/hardcore_random_setup(mob/living/carbon/human/character, antagonist)
+	random_character(TRUE, antagonist)
+	select_hardcore_quirks()
+	hardcore_survival_score = hardcore_survival_score ** 1.2 //30 points would be about 60 score
+
+///Go through all quirks that can be used in hardcore mode and select some based on a random budget.
+/datum/preferences/proc/select_hardcore_quirks()
+
+	var/quirk_budget = rand(8, 35)
+
+
+	all_quirks = list() //empty it out
+
+	var/list/available_hardcore_quirks = SSquirks.hardcore_quirks.Copy()
+
+	while(quirk_budget > 0)
+		for(var/i in available_hardcore_quirks) //Remove from available quirks if its too expensive.
+			var/datum/quirk/T = i
+			if(available_hardcore_quirks[T] > quirk_budget)
+				available_hardcore_quirks -= T
+
+		var/datum/quirk/T = pick(available_hardcore_quirks)
+
+		for(var/V in SSquirks.quirk_blacklist) //Check if the quirk is blacklisted. V is a list
+			var/list/L = V
+			for(var/Q in all_quirks)

```

Comment:
Could you use more descriptive vars than `T`, `V`, `L`, `Q` and the likes?

Reply:
honestly I stole this from the original blacklist code which looks the exact same and I was scared of messing with it because it was hard to understand kek

Reply:
Just change the variable names for some that make sense, it won't affect logic.
It's really much less clear as it is, and understanding what this does is important.

Reply:
yea i was just explaining my reasoning kek, i changed it in the current commit